oath/src/test/java/com/yubico/yubikit/oath/Base32Test.java

```diff
@@ -0,0 +1,102 @@
+        assertEquals("MY======", Base32.encode("f".getBytes()));
+        assertEquals("MZXQ====", Base32.encode("fo".getBytes()));
+        assertEquals("MZXW6===", Base32.encode("foo".getBytes()));
+        assertEquals("MZXW6YQ=", Base32.encode("foob".getBytes()));
+        assertEquals("MZXW6YTB", Base32.encode("fooba".getBytes()));
+        assertEquals("MZXW6YTBOI======", Base32.encode("foobar".getBytes()));
+        assertEquals("PF2WE2LLNF2C243ENMQDELSYFYYCAIBAEE======", Base32.encode("yubikit-sdk 2.X.0   !".getBytes()));
+        assertEquals("KRUGKIDROVUWG2ZAMJZG653OEBTG66BANJ2W24DTEBXXMZLSEB2GQZJANRQXU6JAMRXWOLQ=", Base32.encode("The quick brown fox jumps over the lazy dog.".getBytes()));
+        assertEquals("WZEBZC7I6IQXTNMK", Base32.encode(Codec.fromHex("b6481c8be8f22179b58a")));
+        assertEquals("NG3EQHELVORLMDUPEILZWWGNKY======", Base32.encode(Codec.fromHex("69b6481c8baba2b60e8f22179b58cd56")));
+    }
+
+    @Test
+    public void testDecode() {
+        assertArrayEquals("".getBytes(), Base32.decode(""));
+        assertArrayEquals("f".getBytes(), Base32.decode("MY"));
+        assertArrayEquals("f".getBytes(), Base32.decode("MY======"));
+        assertArrayEquals("fo".getBytes(), Base32.decode("MZXQ"));
+        assertArrayEquals("fo".getBytes(), Base32.decode("MZXQ===="));
+        assertArrayEquals("foo".getBytes(), Base32.decode("MZXW6"));
+        assertArrayEquals("foo".getBytes(), Base32.decode("MZXW6==="));
+        assertArrayEquals("foob".getBytes(), Base32.decode("MZXW6YQ"));
+        assertArrayEquals("foob".getBytes(), Base32.decode("MZXW6YQ="));
+        assertArrayEquals("fooba".getBytes(), Base32.decode("MZXW6YTB"));
+        assertArrayEquals("foobar".getBytes(), Base32.decode("MZXW6YTBOI"));
+        assertArrayEquals("foobar".getBytes(), Base32.decode("MZXW6YTBOI======"));
+        assertArrayEquals("yubikit-sdk 2.X.0   !".getBytes(), Base32.decode("PF2WE2LLNF2C243ENMQDELSYFYYCAIBAEE"));
+        assertArrayEquals("yubikit-sdk 2.X.0   !".getBytes(), Base32.decode("PF2WE2LLNF2C243ENMQDELSYFYYCAIBAEE======"));
+        assertArrayEquals("The quick brown fox jumps over the lazy dog.".getBytes(), Base32.decode("KRUGKIDROVUWG2ZAMJZG653OEBTG66BANJ2W24DTEBXXMZLSEB2GQZJANRQXU6JAMRXWOLQ="));
+        assertArrayEquals(Codec.fromHex("b6481c8be8f22179b58a"), Base32.decode("WZEBZC7I6IQXTNMK"));
+        assertArrayEquals(Codec.fromHex("69b6481c8baba2b60e8f22179b58cd56"), Base32.decode("NG3EQHELVORLMDUPEILZWWGNKY======"));
+    }
+
+    @Test //(expected = IllegalArgumentException.class)
+    public void testDecodeThrows() {
+        try {
+            assertArrayEquals("invalid".getBytes(), Base32.decode("invalidinput"));
+            fail();
+        } catch (IllegalArgumentException ignored) {
+        }
+
+        try {
+            assertArrayEquals("".getBytes(), Base32.decode("M="));
+            fail();
+        } catch (IllegalArgumentException ignored) {
+        }
```

Comment:
What about incomplete padding, should that be an error? For example: `Base32.decode("MY==")`

Or excessive padding? For example: `Base32.decode("MY==============")`

Reply:
I am updating this.